### ISSUE TITLE
Add option to remove buildifier warnings on external repository files

### ISSUE THREAD

**Issue by M-Kerr** [bazelbuild/vscode-bazel#175]

**Comment by thomasvl** [comment]
For clarity, what do you mean by _external repository files_?

Do you mean things not under the folder(s) that VSCode was opened on?
Do you mean things not in the _bazel_ workspace (things pulled in via thinks like http_archive)?

**Comment by M-Kerr** [comment]
Yes things pulled in via rules like http_archive, where the local project is likely not the owner of the code. For example, Google's gtest currently raises a "Keyword parameter "include" for "glob" should be positional. buildifier(keyword-positional-params)" warning in its BUILD file.

It's good to be alerted to these things but it would be helpful to have an option to silence them afterwards.

**Comment by allevato** [comment]
Just to understand the specific problem, is this something that only happens once you open a BUILD file from outside the workspace—the diagnostics get added to the problems view and then don't go away after you close the file?

If that's the case, I think it would make sense to either (1) not add the diagnostics to the problem view in the first place for files that aren't in the workspace (just show them in the editor) or (2) add them but remove them once the document is closed. I don't have a strong opinion about which of those is better.
